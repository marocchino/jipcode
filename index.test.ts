import { locate, prefectureCode } from "./index";

describe("prefectureCode", () => {
  it("returns prefecture code hash", () => {
    expect(prefectureCode).toMatchSnapshot();
  });
});

describe("locate", () => {
  it("returns nil array when long number", async () => {
    expect(await locate("100000000")).toEqual([]);
  });

  it("returns nil array when not number", async () => {
    expect(await locate("ABCDEFG")).toEqual([]);
  });

  it("returns nil array when not exists", async () => {
    expect(await locate("0081234")).toEqual([]);
  });

  it("returns when with hyphen", async () => {
    expect(await locate("115-0051")).toEqual([
      { city: "北区", prefecture: "東京都", town: "浮間", zipcode: "1150051" },
    ]);
  });

  it("returns when no options", async () => {
    expect(await locate("1150051")).toEqual([
      { city: "北区", prefecture: "東京都", town: "浮間", zipcode: "1150051" },
    ]);
  });

  it("returns when prefectureCode options", async () => {
    expect(await locate("1150051", { prefectureCode: true })).toEqual([
      {
        city: "北区",
        prefecture: "東京都",
        prefectureCode: 13,
        town: "浮間",
        zipcode: "1150051",
      },
    ]);
  });
});
