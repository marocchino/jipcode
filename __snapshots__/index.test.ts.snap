// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prefectureCode returns prefecture code hash 1`] = `
Object {
  "三重県": 24,
  "京都府": 26,
  "佐賀県": 41,
  "兵庫県": 28,
  "北海道": 1,
  "千葉県": 12,
  "和歌山県": 30,
  "埼玉県": 11,
  "大分県": 44,
  "大阪府": 27,
  "奈良県": 29,
  "宮城県": 4,
  "宮崎県": 45,
  "富山県": 16,
  "山口県": 35,
  "山形県": 6,
  "山梨県": 19,
  "岐阜県": 21,
  "岡山県": 33,
  "岩手県": 3,
  "島根県": 32,
  "広島県": 34,
  "徳島県": 36,
  "愛媛県": 38,
  "愛知県": 23,
  "新潟県": 15,
  "東京都": 13,
  "栃木県": 9,
  "沖縄県": 47,
  "滋賀県": 25,
  "熊本県": 43,
  "石川県": 17,
  "神奈川県": 14,
  "福井県": 18,
  "福岡県": 40,
  "福島県": 7,
  "秋田県": 5,
  "群馬県": 10,
  "茨城県": 8,
  "長崎県": 42,
  "長野県": 20,
  "青森県": 2,
  "静岡県": 22,
  "香川県": 37,
  "高知県": 39,
  "鳥取県": 31,
  "鹿児島県": 46,
}
`;
